lib/src/values/logic_value.dart

```diff
@@ -699,7 +699,11 @@ abstract class LogicValue {
       return LogicValue.filled(other.width, LogicValue.x);
     }
 
-    if (this is _BigLogicValue || other is BigInt || other is _BigLogicValue) {
+    if (this is _BigLogicValue ||
+        other is BigInt ||
+        other is _BigLogicValue ||
+        (this is _FilledLogicValue) && this.width >= 64 ||
```

Comment:
I don't think this should be `>=`, just `>`.  However, it appears there's a separate signed bug related to when the value is equal to 64 that you've discovered.  You could file it as a separate bug or just include some description of it added in the PR.

I thought a fix could be to use [`toUnsigned`](https://api.flutter.dev/flutter/dart-core/int/toUnsigned.html) in the `toInt` functions for LogicValue implementations, but it seems like that's limited to 63 bits.  So perhaps the `>=` is ok for the `_doCompare`, but the `_doMath` should support all 64 bits.  We still should add some additional tests around the fix to make sure it actually works for 63, 64, and >64 bit numbers (including ones that could be interpreted as negative) for both comparison and math.